{{ cookiecutter.__package_name_kebab_case }}/.gitlab-ci.yml

```
@@ -1,103 +1,81 @@
 stages:
-  - prebuild
   - build
   - test
   - {% if cookiecutter.with_fastapi_api|int or cookiecutter.with_streamlit_app|int %}deploy{% else %}publish{% endif %}
 
-Compute CI image hash:
-  stage: prebuild
-  script:
-    - export DOCKER_IMAGE_SHA="$(sha1sum Dockerfile poetry.lock pyproject.toml | sha1sum | cut -c 1-12)"
-    - echo "CI_IMAGE_SHA=$DOCKER_IMAGE_SHA" >> .env
-  artifacts:
-    reports:
-      dotenv: .env
+variables:
+  DOCKER_TLS_CERTDIR: "/certs"
+
+.python_matrix:
+  parallel:
+    matrix:
+      - PYTHON_VERSION: ["{{ cookiecutter.python_version }}"]
 
-# Base Docker build script.
-.docker:
-  image: docker:stable
+.install_devcontainers_cli:
+  cache:
+    paths:
+      - .apk_cache
+      - .npm_cache
+  before_script:
+    - mkdir -p .apk_cache && apk add --cache-dir .apk_cache npm
```

I don't think we should remove the `--update-cache` flag here, otherwise this will fail the first time we install it as the APK cache won't be up to date in the project cache. I suppose you want to avoid fetching the APK package list. What we could do is something like this:


```suggestion
    - mkdir -p .apk_cache && (apk add --cache-dir .apk_cache npm || apk add --update-cache --cache-dir .apk_cache npm)
```

But that may be overkill 🙂

---

I actually tried running `mkdir -p .apk_cache && apk add --cache-dir .apk_cache npm` in a fresh Docker container before making this change and it ran successfully. Am I missing something?

---

Weird, I could swear I've been bitten by that before, but it just works so that's great!